### ISSUE TITLE
Minion Rework Idea

### ISSUE THREAD

**Issue by NeoVolt** [Tslat/Advent-Of-Ascension#1696]
**Minion Rework**

Creation is a skill, that allows you to make minions, but it's always felt a bit bland. So, here's my idea to improve it, or more specifically, the minions:

**Mechanics:**
Make Minion Slabs non-consumable. Instead, they will use durability. I'd say make it range from 64-256 depending on what level is required to use the slab, with the higher durability ones needing a higher level.
To balance this out, of course, Minion Slabs will require more expensive materials to craft. (Ex. penguins might require Snow Blocks, Packed Ice and a type of fish.)

Adding a cap to the maximum amount of minions that can be spawned would be good for balance. This would be used to balance out Minion Slabs so you can't just create an army. I would say that this cap should start at 3, and every 20 Creation levels it should increase by 1.

**More Variety**

Add more unique types of minions. Currently, (as far as I'm aware) they all share the exact same AI, run up to something and punch it, but some have some sort of special effect. But what if we had more variety than special effects, like flying or ranged minions? Maybe special minions you can only spawn one of at a time that are extra powerful? There's quite a few options out there, and it would be great to see some variety in them.

Along with more types of minions, more types of Creation-based armor would be good too. Currently, Creation only has 1 armor set, which can only be worn at 100 Creation (Unless you want to count Commander armor, in which case it's 2) Perhaps some armor that increased minion's health, allowed them to naturally regenerate, or increased your maximum minions you can spawn by a few would be nice.

**Enchantments**

If Minion Slabs aren't consumable, they should be able to have enchantments like other items that have durability. While the basic 2 enchantments every enchantable item can get will be applicable (Unbreaking and Mending), I think there should be some new ones that can only be used on these Slabs. Here's some I've some up with:

Discount I - III: Takes 10% less creation points per level to spawn this minion. Can be gotten through an enchantment table.

Salvage I - III: Restore 20% of the creation points per level required to spawn this minion when the minion dies. (Note: If you have discount as well, this will decrease the amount restored) Can be gotten through an enchantment table.

Guard I: Negates all damage from the first attack the minion gets hit by. Cannot be gotten through an enchantment table.

**TL;DR**
Make Slabs be non-consumable and use durability, add a spawn cap for minions, make some more unique minions and creation-based armor, and allow for enchantments on slabs.
